doc/source/whatsnew/v0.23.0.txt

```diff
@@ -363,6 +363,7 @@ Reshaping
 - Bug in :func:`Series.rank` where ``Series`` containing ``NaT`` modifies the ``Series`` inplace (:issue:`18521`)
 - Bug in :func:`cut` which fails when using readonly arrays (:issue:`18773`)
 - Bug in :func:`Dataframe.pivot_table` which fails when the ``aggfunc`` arg is of type string.  The behavior is now consistent with other methods like ``agg`` and ``apply`` (:issue:`18713`)
+- Bug in :func:`pandas.core.dtypes.concat._get_series_result_type` which returns SparseDataFrame even if not all contained in Frame are sparse. (:issue:`18914` and :issue:`18686`)
```

Comment:
the whatsnew note should be from a user perspective (e.g. the error that happens with get_dummies).

Reply:
I changed it to be about pd.concat I think that's user facing right? While get_dummies is the symptom pd.concat is the real problem. Lemme know if you want me to change that thanks!